plumbing/protocol/version.go

```diff
@@ -0,0 +1,51 @@
+package protocol
+
+import (
+	"errors"
+	"fmt"
+)
+
+var ErrUnknownProtocol = errors.New("unknown Git Wire protocol")
+
+// Version sets the preferred version for the Git wire protocol.
+type Version int
+
+const (
+	Unknown Version = -1
```

Comment:
```suggestion
	Undefined Version = -1
```
Perhaps `Undefined` signifies the intention better